### DISCUSSION TITLE
手动添加自动代理规则网址，请问应该编辑哪个文件？

### DISCUSSION THREAD

**Discussion by jzmswe** [SeaHOH/GotoX#105] [问答/咨询] [answered]
一直想问这个问题，哪位朋友指点一下，多谢！

**Comment by SeaHOH** [comment, marked as answer]
如果是 Windows，你应该能在托盘工具菜单中看到才对。
或者挨个打开 config 文件夹的 ini 文件，你会看到详细的配置说明。

只使用默认配置，你就错过了许多快速解决网络访问问题的机会，
尤其是现在 GAE 代理不好用、又没有其它代理的情况下。

**Comment by jzmswe** [comment]
@SeaHOH 把
.nyt.com
.nytimes.com
.cloudfront.net
.nytcn.me
加入规则后，这个链接中的图片还是无法显示：
https://cn.nytimes.com/slideshow/20180507/c08-met-gala-redcarpet-ss/?utm_source=mostviewed-daily&utm_medium=cpc&utm_campaign=mostviewed

搞定了，把 .cloudfront.net 清除就OK了，多谢老大指点！

自动代理规则里基本没几个，看来以后还是主要用全局代理。

**Comment by SeaHOH** [comment]
主要还是用 DoH 反污染（另外安装工具也行）和伪造 SNI 来直连，这样就基本上不用其它代理了。
就是个别网站速度和连通率不是很理想。

**Comment by Just2co** [comment]
@SeaHOH 海少，伪造SNI通讯已经成熟可用了？这在当前大部分我们要上的网站都已经升级成了HTTPS的情况下算是利好啊

**Comment by SeaHOH** [comment]
https://hikinggfw.org/blocked_sites
这里有封锁情况统计，如果是 IP 被封就没法直连。会有以下例外：
1. GFW 只封了 IPv4，这种大机率可以通过伪造 SNI 直连 IPv6。
1. 网站只提供 IPv4，可以尝试使用 DNS64/NAT64 解析出可用的 IPv6，甚至可能只需把此 IP 设置成 `forward` 规则，而不用伪造 SNI，。

**Comment by Just2co** [comment]
@SeaHOH 好的，有空研究下，你现在已经放弃GAE方式了吗？毕竟还是有些IP是可用的啊，虽然大幅减少

**Comment by SeaHOH** [comment]
不会移除 GAE 代理，会再增加一个自定义域名支持。
然后就没了，估计 GAE 方面也不会再有什么大变化了，也不会添加其它代理。
GotoX 就只是个调度和修改工具，可以自己添加规则。

**Comment by SeaHOH** [comment]
> 自定义域名支持 什么时候可以实现？

还在脑袋里整理要改的地方，感觉有点多。。。所以不会很快。

---

已支持
